File: packages/frontend/src/services/TeamService.js
// File: frontend/src/services/TeamService.js
import axios from 'axios';

const API_URL = '/api/teams';

/**
 * Service for team-related API calls.
 */
class TeamService {
  /**
   * Get all teams with pagination.
   * 
   * @param {Object} pageable - Pagination parameters
   * @returns {Promise} - Promise with response data
   */
  getAllTeams(pageable = { page: 0, size: 20 }) {
    return axios.get(API_URL, { params: pageable });
  }

  /**
   * Get a team by ID.
   * 
   * @param {number} id - Team ID
   * @returns {Promise} - Promise with response data
   */
  getTeamById(id) {
    return axios.get(`${API_URL}/${id}`);
  }

  /**
   * Get a team by name.
   * 
   * @param {string} name - Team name
   * @returns {Promise} - Promise with response data
   */
  getTeamByName(name) {
    return axios.get(`${API_URL}/name/${name}`);
  }

  /**
   * Create a new team.
   * 
   * @param {Object} teamData - Team data
   * @returns {Promise} - Promise with response data
   */
  createTeam(teamData) {
    return axios.post(API_URL, teamData);
  }

  /**
   * Update a team.
   * 
   * @param {number} id - Team ID
   * @param {Object} teamData - Updated team data
   * @returns {Promise} - Promise with response data
   */
  updateTeam(id, teamData) {
    return axios.put(`${API_URL}/${id}`, teamData);
  }

  /**
   * Delete a team.
   * 
   * @param {number} id - Team ID
   * @returns {Promise} - Promise with response data
   */
  deleteTeam(id) {
    return axios.delete(`${API_URL}/${id}`);
  }

  /**
   * Get teams by type.
   * 
   * @param {string} type - Team type
   * @returns {Promise} - Promise with response data
   */
  getTeamsByType(type) {
    return axios.get(`${API_URL}/type/${type}`);
  }

  /**
   * Get active teams.
   * 
   * @returns {Promise} - Promise with response data
   */
  getActiveTeams() {
    return axios.get(`${API_URL}/active`);
  }

  /**
   * Get teams by department.
   * 
   * @param {string} department - Department name
   * @returns {Promise} - Promise with response data
   */
  getTeamsByDepartment(department) {
    return axios.get(`${API_URL}/department/${department}`);
  }

  /**
   * Get teams by functional area.
   * 
   * @param {string} functionalArea - Functional area
   * @returns {Promise} - Promise with response data
   */
  getTeamsByFunctionalArea(functionalArea) {
    return axios.get(`${API_URL}/functional-area/${functionalArea}`);
  }

  /**
   * Get teams managed by a specific user.
   * 
   * @param {number} managerId - Manager's user ID
   * @returns {Promise} - Promise with response data
   */
  getTeamsByManager(managerId) {
    return axios.get(`${API_URL}/manager/${managerId}`);
  }

  /**
   * Get teams that a user belongs to.
   * 
   * @param {number} userId - User ID
   * @returns {Promise} - Promise with response data
   */
  getTeamsByMember(userId) {
    return axios.get(`${API_URL}/member/${userId}`);
  }

  /**
   * Add a member to a team.
   * 
   * @param {number} teamId - Team ID
   * @param {number} userId - User ID to add
   * @returns {Promise} - Promise with response data
   */
  addMemberToTeam(teamId, userId) {
    return axios.put(`${API_URL}/${teamId}/members/${userId}`);
  }

  /**
   * Remove a member from a team.
   * 
   * @param {number} teamId - Team ID
   * @param {number} userId - User ID to remove
   * @returns {Promise} - Promise with response data
   */
  removeMemberFromTeam(teamId, userId) {
    return axios.delete(`${API_URL}/${teamId}/members/${userId}`);
  }

  /**
   * Set the manager of a team.
   * 
   * @param {number} teamId - Team ID
   * @param {number} managerId - Manager's user ID
   * @returns {Promise} - Promise with response data
   */
  setTeamManager(teamId, managerId) {
    return axios.put(`${API_URL}/${teamId}/manager/${managerId}`);
  }

  /**
   * Activate a team.
   * 
   * @param {number} teamId - Team ID
   * @returns {Promise} - Promise with response data
   */
  activateTeam(teamId) {
    return axios.put(`${API_URL}/${teamId}/activate`);
  }

  /**
   * Deactivate a team.
   * 
   * @param {number} teamId - Team ID
   * @returns {Promise} - Promise with response data
   */
  deactivateTeam(teamId) {
    return axios.put(`${API_URL}/${teamId}/deactivate`);
  }

  /**
   * Search teams by name or description.
   * 
   * @param {string} query - Search query
   * @param {Object} pageable - Pagination parameters
   * @returns {Promise} - Promise with response data
   */
  searchTeams(query, pageable = { page: 0, size: 20 }) {
    return axios.get(`${API_URL}/search`, {
      params: {
        query,
        ...pageable
      }
    });
  }
}

export default new TeamService();

File: packages/backend/src/main/java/com/enterprise/core/team/service/TeamService.java
// File: backend/src/main/java/com/enterprise/core/team/service/TeamService.java
package com.enterprise.core.team.service;

import com.enterprise.core.team.dto.TeamCreateDto;
import com.enterprise.core.team.dto.TeamDto;
import com.enterprise.core.team.dto.TeamUpdateDto;
import com.enterprise.core.team.entity.TeamType;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

/**
 * Service interface for team management operations.
 */
public interface TeamService {

    /**
     * Create a new team.
     *
     * @param teamCreateDto The team creation data
     * @return The created team
     */
    TeamDto createTeam(TeamCreateDto teamCreateDto);

    /**
     * Get a team by ID.
     *
     * @param id The team ID
     * @return The team if found
     */
    TeamDto getTeamById(Long id);

    /**
     * Get a team by name.
     *
     * @param name The team name
     * @return The team if found
     */
    TeamDto getTeamByName(String name);

    /**
     * Update a team.
     *
     * @param id The team ID
     * @param teamUpdateDto The updated team data
     * @return The updated team
     */
    TeamDto updateTeam(Long id, TeamUpdateDto teamUpdateDto);

    /**
     * Delete a team.
     *
     * @param id The team ID
     */
    void deleteTeam(Long id);

    /**
     * Get all teams with pagination.
     *
     * @param pageable Pagination information
     * @return Page of teams
     */
    Page<TeamDto> getAllTeams(Pageable pageable);

    /**
     * Get teams by type.
     *
     * @param type The team type
     * @return List of teams
     */
    List<TeamDto> getTeamsByType(TeamType type);

    /**
     * Get active teams.
     *
     * @return List of active teams
     */
    List<TeamDto> getActiveTeams();

    /**
     * Get teams by department.
     *
     * @param department The department
     * @return List of teams
     */
    List<TeamDto> getTeamsByDepartment(String department);

    /**
     * Get teams by functional area.
     *
     * @param functionalArea The functional area
     * @return List of teams
     */
    List<TeamDto> getTeamsByFunctionalArea(String functionalArea);

    /**
     * Get teams managed by a specific user.
     *
     * @param managerId The manager's user ID
     * @return List of teams
     */
    List<TeamDto> getTeamsByManager(Long managerId);

    /**
     * Get teams that a user belongs to.
     *
     * @param userId The user ID
     * @return List of teams
     */
    List<TeamDto> getTeamsByMember(Long userId);

    /**
     * Add a member to a team.
     *
     * @param teamId The team ID
     * @param userId The user ID to add
     * @return The updated team
     */
    TeamDto addMemberToTeam(Long teamId, Long userId);

    /**
     * Remove a member from a team.
     *
     * @param teamId The team ID
     * @param userId The user ID to remove
     * @return The updated team
     */
    TeamDto removeMemberFromTeam(Long teamId, Long userId);

    /**
     * Set the manager of a team.
     *
     * @param teamId The team ID
     * @param managerId The manager's user ID
     * @return The updated team
     */
    TeamDto setTeamManager(Long teamId, Long managerId);

    /**
     * Activate a team.
     *
     * @param teamId The team ID
     * @return The updated team
     */
    TeamDto activateTeam(Long teamId);

    /**
     * Deactivate a team.
     *
     * @param teamId The team ID
     * @return The updated team
     */
    TeamDto deactivateTeam(Long teamId);

    /**
     * Search teams by name or description.
     *
     * @param query The search query
     * @param pageable Pagination information
     * @return Page of teams
     */
    Page<TeamDto> searchTeams(String query, Pageable pageable);
}